### PR TITLE
VDS-151: Add blades with customer properties

### PR DESCRIPTION
**Pull request by ilyawzrd**
Снимок экрана 2020-11-24 в 09 55 38

### CODE REVIEW

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js [hunk 1]
```diff
@@ -33,6 +33,21 @@ angular.module('virtoCommerce.demoSolutionFeaturesModule')
...
+            $scope.createCustomerFilter = function () {
+                var ruleCreationBlade = {
+                    id: "createCustomerSegmentRule",
+                    controller: 'virtoCommerce.demoSolutionFeaturesModule.ruleCreationController',
+                    title: 'demoSolutionFeaturesModule.blades.customer-segments-rule-creation.title',
+                    subtitle: 'demoSolutionFeaturesModule.blades.customer-segments-rule-creation.subtitle',
+                    template: 'Modules/$(virtoCommerce.DemoSolutionFeaturesModule)/Scripts/blades/rule-creation.tpl.html',
+                    // onSelected: function (editedProperties) {
+                    // TODO: save the rule
+                    // }
```
Comment (asvishnyakov) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js]: Remove commented out code

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/rule-creation.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                            // executeMethod: (pickingBlade) => {
+                            //     var viewerBlade = {
+                            //         id: 'propertiesSelector',
+                            //         controller: 'virtoCommerce.demoSolutionFeaturesModule.dynamicAssociationViewerController',
+                            //         template: 'Modules/$(virtoCommerce.demoSolutionFeaturesModule)/Scripts/blades/dynamicAssociation-viewer.tpl.html',
+                            //         properties: pickingBlade.currentEntities
+                            //     };
+                            //     bladeNavigationService.showBlade(viewerBlade, pickingBlade);
+                            // },
+                            //canExecuteMethod: () => true
```
Comment (asvishnyakov) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/rule-creation.js]: Remove commented out code

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js (hunk 1 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-detail.js]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/rule-creation.js (hunk 2 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/rule-creation.js]: done

**Review by asvishnyakov — CHANGES_REQUESTED** [review]
I'll prefer use `customerSegment-` prefix, i.e.
customerSegment-detail
customerSegment-rule
customerSegment-properties
customerSegment-property-values
@pushnitsa What's your opinion?

**Comment by pushnitsa** [comment]
@ilyawzrd Could you please rename files with Aleksander's proposed prefix?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                    id: 'propertiesEditor',
+                    title: 'demoSolutionFeaturesModule.blades.customer-segments-rule-creation.select-properties-blade-title',
+                    headIcon: 'fa-pie-chart',
+                    controller: 'virtoCommerce.demoSolutionFeaturesModule.editPropertiesController',
+                    template: 'Modules/$(virtoCommerce.DemoSolutionFeaturesModule)/Scripts/blades/customerSegment-property-values.tpl.html',
+                    entities: blade.editedProperties,
+                    onSelected: function (editedProps) {
+                        blade.editedProperties = editedProps;
+                    },
+                    toolbarCommands: [
```
Comment (asvishnyakov) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.js]: Move it to `virtoCommerce.demoSolutionFeaturesModule.editPropertiesController`, i.e.
```js
var blade = $scope.blade;
blade.toolbarCommands = ...
```

**Review by asvishnyakov — CHANGES_REQUESTED** [review]
Shouldn't blade controller names correspond file names?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.js (hunk 3 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.js]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js [hunk 4]
```diff
@@ -15,15 +14,13 @@ angular.module('virtoCommerce.demoSolutionFeaturesModule')
...
-            $scope.setForm = (form) => { formScope = form };
-
             $scope.mainParameters = function () {
                 const parametersBlade = {
                     id: "mainParameters",
                     title: "demoSolutionFeaturesModule.blades.customer-segments-parameters.title",
                     subtitle: 'demoSolutionFeaturesModule.blades.customer-segments-parameters.subtitle',
-                    controller: 'virtoCommerce.demoSolutionFeaturesModule.customerSegmentsParametersController',
-                    template: 'Modules/$(virtoCommerce.DemoSolutionFeaturesModule)/Scripts/blades/mainParameters.tpl.html',
+                    controller: 'virtoCommerce.demoSolutionFeaturesModule.customerSegmentsMainParametersController',
```
Comment (asvishnyakov) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js]: Singular, not plural. Here and below
`customerSegments` → `customerSegment`

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-properties.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
+angular.module('virtoCommerce.demoSolutionFeaturesModule')
+    .controller('virtoCommerce.demoSolutionFeaturesModule.propertiesController', ['$scope', 'platformWebApp.bladeNavigationService', function ($scope, bladeNavigationService) {
```
Comment (asvishnyakov) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-properties.js]: customerSegmentPropertiesController?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-property-values.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+angular.module('virtoCommerce.demoSolutionFeaturesModule')
+.controller('virtoCommerce.demoSolutionFeaturesModule.propertyValuesController', ['$scope', 'platformWebApp.bladeNavigationService', 'platformWebApp.dynamicProperties.dictionaryItemsApi', function ($scope, bladeNavigationService, dictionaryItemsApi) {
```
Comment (asvishnyakov) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-property-values.js]: customerSegmentPropertyValuesController?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
+angular.module('virtoCommerce.demoSolutionFeaturesModule')
+    .controller('virtoCommerce.demoSolutionFeaturesModule.ruleController', [
```
Comment (asvishnyakov) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.js]: customerSegmentRuleController?

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.tpl.html [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<div class="blade-static __bottom">
+    <button class="btn" ng-click="saveRule()" ng-disabled="!canSave()">
+        <span>Apply</span>
```
Comment (asvishnyakov) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.tpl.html]: Please move to localization resource and change to "Confirm" (discussed with Maya and @pushnitsa)

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js (hunk 4 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-properties.js (hunk 5 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-properties.js]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.js (hunk 7 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.js]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-property-values.js (hunk 6 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-property-values.js]: done

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.tpl.html (hunk 8 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-rule.tpl.html]: done

**Review by asvishnyakov — APPROVED** [review]

**Review by pushnitsa — APPROVED** [review]

**Pull request merged**